dvc/tree/webdav.py

```diff
@@ -216,31 +216,44 @@ def _download(self, from_info, to_file, name=None, no_progress_bar=False):
                 )
 
     # Uploads file to remote
-    def _upload(self, from_file, to_info, name=None, no_progress_bar=False):
+    def _upload(
+        self,
+        from_file,
+        to_info,
+        name=None,
+        no_progress_bar=False,
+        chunked=False,
+    ):
         # First try to create parent directories
         self.makedirs(to_info.parent)
 
-        # Progress from HTTPTree
-        def chunks():
-            with open(from_file, "rb") as fd:
-                with Tqdm.wrapattr(
-                    fd,
-                    "read",
-                    total=None
-                    if no_progress_bar
-                    else os.path.getsize(from_file),
-                    leave=False,
-                    desc=to_info.url if name is None else name,
-                    disable=no_progress_bar,
-                ) as fd_wrapped:
-                    while True:
-                        chunk = fd_wrapped.read(self.CHUNK_SIZE)
-                        if not chunk:
-                            break
-                        yield chunk
-
-        # Upload to WebDAV via buffer
-        self._client.upload_to(buff=chunks(), remote_path=to_info.path)
+        if chunked:
+            # Progress from HTTPTree
+            def chunks():
+                with open(from_file, "rb") as fd:
+                    with Tqdm.wrapattr(
+                        fd,
+                        "read",
+                        total=None
+                        if no_progress_bar
+                        else os.path.getsize(from_file),
+                        leave=False,
+                        desc=to_info.url if name is None else name,
+                        disable=no_progress_bar,
+                    ) as fd_wrapped:
+                        while True:
+                            chunk = fd_wrapped.read(self.CHUNK_SIZE)
+                            if not chunk:
+                                break
+                            yield chunk
+
+            # Upload to WebDAV via buffer
+            self._client.upload_to(buff=chunks(), remote_path=to_info.path)
+        else:
+            # Upload to WebDAV without chunking
+            self._client.upload_file(
```

Comment:
@LucaButera, this won't show up the progress bar. Can you check the snippets on the issue with progress bar? Please see if works.

Reply:
From my testing the progress bar does work, but instead of having a global progress bar with "sub-bars" for each file you only get a single bar which updates for each file sent (more or less). This is perfectly fine for uploads of folders containing many reasonably sized files. It may actually become a problem when you upload one (or more) huge files. In such cases you have no information about the progress.

Reply:
Can you check with the snippet here that I mentioned: https://github.com/iterative/dvc/issues/4796#issuecomment-719181936?

Reply:
Here's what I found out:
The snippet works flawlessly. It gives per file progress bar which updates correctly even in the case of a single, really big, file.

A few things that may or may not be related that I wish to mention:
* When querying the remote cache the progress bar starts from ```0/512``` even though the files are more than that. In fact, after it reaches 512, it starts going up like ```513/?``` and so on.
* After pushing a 7GB file, cache querying became extremely slow. I am not sure if it is just because the cache itself was bigger. Previously my remote contained 2.9GB divided in ~900 files and it seemed insanely fast in proportion.
* During push some files generate an error 403 or 423. Again this disappears the second time you try to push the files that gave the error.

Let me know what you think about this and how to proceed, thank you!

Reply:
@skshetry Did you have time to read the issues I mentioned? Do you think they are just server related problems or could there be some issues with dvc?

Reply:
@LucaButera, I am not sure of 1. 

For the 2, can you post the log? This patch should not affect that. Maybe, you were uploading the folders/files too-many times during the tests, that the local cache was getting invalidated. We try to guess the number of files to reduce no. of requests, so, that's my guess. Maybe, same thing is affecting 1.

For 3, it might happen with Owncloud/NextCloud if you are deleting/adding files too quickly, re in your test. So, it's not a DVC's issue. If that happens, retry again in a while.

Anyway, it should not be related to this patch. Please create a issue if that happens later.